diff --git a/scipy/io/matlab/_mio5_params.py b/scipy/io/matlab/_mio5_params.py
index 0d60b8e7a..2ad2b90fe 100644
--- a/scipy/io/matlab/_mio5_params.py
+++ b/scipy/io/matlab/_mio5_params.py
@@ -278,4 +278,4 @@ class MatlabOpaque(np.ndarray):
 
 
 OPAQUE_DTYPE = np.dtype(
-    [('s0', 'O'), ('s1', 'O'), ('s2', 'O'), ('arr', 'O')])
+    [('_TypeSystem', 'O'), ('_Class', 'O'), ('_ObjectMetadata', 'O')])
diff --git a/scipy/io/matlab/_mio5_utils.pyx b/scipy/io/matlab/_mio5_utils.pyx
index b7abd7104..0cf2708dd 100644
--- a/scipy/io/matlab/_mio5_utils.pyx
+++ b/scipy/io/matlab/_mio5_utils.pyx
@@ -596,7 +596,7 @@ cdef class VarReader5:
         # all miMATRIX types except the mxOPAQUE_CLASS have dims and a
         # name.
         if mc == mxOPAQUE_CLASS:
-            header.name = None
+            header.name = self.read_int8_string()
             header.dims = None
             return header
         header.n_dims = self.read_into_int32s(header.dims_ptr, sizeof(header.dims_ptr))
@@ -986,8 +986,7 @@ cdef class VarReader5:
         # Cython (0.23.4).
         res = np.empty((1,), dtype=OPAQUE_DTYPE)
         res0 = res[0]
-        res0['s0'] = self.read_int8_string()
-        res0['s1'] = self.read_int8_string()
-        res0['s2'] = self.read_int8_string()
-        res0['arr'] = self.read_mi_matrix()
+        res0['_TypeSystem'] = PyUnicode_FromString(self.read_int8_string())
+        res0['_Class'] = PyUnicode_FromString(self.read_int8_string())
+        res0['_ObjectMetadata'] = self.read_mi_matrix()
         return res
